lib/phase/flatten.js

```diff
@@ -2,14 +2,41 @@
 
 const Hoek = require('hoek');
 const clone = require('clone');
+const TEMPLATE_REGEX = require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME;
+const NAME_MATCH = 1;
+const VERSION_MATCH = 3;
+const LABEL_MATCH = 7;
 
 /**
- * Overlays the stuff specified in a specific job on top of the defaults in shared
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: individual job settings vs shared settings
+ * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
+ * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
+ */
+function merge(newJob, oldJob) {
+    // Intialize new job with default fields
+    newJob.environment = newJob.environment || {};
+    newJob.settings = newJob.settings || {};
+
+    // Merge
+    Object.assign(newJob.environment, oldJob.environment || {});
+    Object.assign(newJob.settings, oldJob.settings || {});
+    newJob.image = oldJob.image || newJob.image;
+
+    // Merge secrets
+    const newSecrets = newJob.secrets || [];
+    const oldSecrets = oldJob.secrets || [];
+
+    newJob.secrets = [...new Set([...newSecrets, ...oldSecrets])];  // remove duplicate
+}
+
+/**
+ * Overlays the fields specified in a specific job on top of the defaults in shared
  *
  * @method flattenSharedIntoJobs
  * @param  {Job}          shared A kind of default Job template
- * @param  {Object}       jobs   List of jobs (name => job)
- * @return {Object}              Updated list of jobs after merging
+ * @param  {Object}       jobs   Object with all the jobs
+ * @return {Object}              Updated object with jobs after merging
```

Comment:
Does it modify the `jobs` object? It would be more clear if it described that it returns a new object and not a modified `jobs` object.